### ISSUE TITLE
Not starting due: module.js:327 throw err; Error: Cannot find module ... zookeeper.node

### ISSUE THREAD

**Issue by bambrozio** [yfinkelstein/node-zookeeper#129]
I've just cloned the project, and I'm trying to perform the example.js, but it's not working, and I could not figure out the problem... Did I forget something?
Here goes my attempts to fix it:
`
# node example.js 
module.js:327
    throw err;
    ^

Error: Cannot find module '/home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper-master/lib/../build/zookeeper.node'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper-master/lib/zookeeper.js:6:16)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
# ls -l
total 68
-rwxrwxrwx 1 bruno bruno  2270 Jun 25 22:20 binding.gyp
drwxrwxrwx 3 bruno bruno  4096 Sep  9 18:26 build
drwxrwxrwx 2 bruno bruno  4096 Jun 25 22:20 deps
-rwxrwxrwx 1 bruno bruno   787 Sep  9 18:23 example.js
drwxrwxrwx 2 bruno bruno  4096 Jun 25 22:20 lib
-rwxrwxrwx 1 bruno bruno  1086 Jun 25 22:20 LICENSE-MIT.txt
drwxr-xr-x 7 root  root   4096 Sep 12 10:33 node_modules
-rwxrwxrwx 1 bruno bruno  1130 Jun 25 22:20 package.json
drwxrwxrwx 2 bruno bruno  4096 Jun 25 22:20 patches
-rwxrwxrwx 1 bruno bruno 14540 Jun 25 22:20 README.md
drwxrwxrwx 2 bruno bruno  4096 Jun 25 22:20 scripts
drwxrwxrwx 2 bruno bruno  4096 Jun 25 22:20 src
drwxrwxrwx 2 bruno bruno  4096 Sep  9 17:53 test
# node-gyp -v
v3.0.3
# node -v
v4.5.0
# npm -v
2.15.9
# npm install
npm WARN package.json zookeeper@3.4.8-2 No license field.
npm WARN cannot run in wd zookeeper@3.4.8-2 node-gyp rebuild (wd=/home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper-master)
npm WARN cannot run in wd zookeeper@3.4.8-2 ./scripts/prepublish.sh (wd=/home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper-master)
async@0.2.10 node_modules/async

underscore@1.8.3 node_modules/underscore

webworker@0.8.4 node_modules/webworker

nan@2.3.5 node_modules/nan

log4js@0.6.38 node_modules/log4js
├── semver@4.3.6
└── readable-stream@1.0.34 (isarray@0.0.1, string_decoder@0.10.31, inherits@2.0.3, core-util-is@1.0.2)
root@dockerfordevops:/home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper-master# node example.js 
module.js:327
    throw err;
    ^

Error: Cannot find module '/home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper-master/lib/../build/zookeeper.node'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper-master/lib/zookeeper.js:6:16)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)

# node-gyp configure build
gyp info it worked if it ends with ok
gyp info using node-gyp@3.0.3
gyp info using node@4.2.6 | linux | x64
gyp info spawn python2
gyp info spawn args [ '/usr/share/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper-master/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/share/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/include/nodejs/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/usr/include/nodejs',
gyp info spawn args   '-Dnode_gyp_dir=/usr/share/node-gyp',
gyp info spawn args   '-Dnode_lib_file=node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper-master',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
module.js:327
    throw err;
    ^

Error: Cannot find module 'nan'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at [eval]:1:1
    at Object.exports.runInThisContext (vm.js:54:17)
    at Object.<anonymous> ([eval]-wrapper:6:22)
    at Module._compile (module.js:409:26)
    at node.js:579:27
    at nextTickCallbackWith0Args (node.js:420:9)
gyp: Call to 'node -e "require('nan')"' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/share/node-gyp/lib/configure.js:354:16)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 4.4.0-36-generic
gyp ERR! command "/usr/bin/nodejs" "/usr/bin/node-gyp" "configure" "build"
gyp ERR! cwd /home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper-master
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok`

**Comment by bambrozio** [comment]
btw:
`# uname -a
Linux dockerfordevops 4.4.0-36-generic #55-Ubuntu SMP Thu Aug 11 18:01:55 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
`

**Comment by bambrozio** [comment]
btw:
`
# uname -a
Linux dockerfordevops 4.4.0-36-generic #55-Ubuntu SMP Thu Aug 11 18:01:55 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
`

**Comment by mdlavin** [comment]
One thing that stands out as odd to me is your `node_modules` project is owned by `root`. I wouldn't expect that, and it might cause the `cannot run in wd` warning. Can you remove your `node_modules` directory and try to `npm install` as your normal user?

**Comment by mdlavin** [comment]
Any luck removing the root owned folders?

**Comment by bambrozio** [comment]
Hello,
I've just added root, and performed everything as root too, in a attempt to solve the issue. But actually I figure it out not:

As I was building locally, instead of just using it, I had to add the ZooKeeper at:
/home/bruno/github.ibm.com/orgs/BAmbrozi/zk-client-poc/node-zookeeper/deps/zookeeper

Then I could perform the npm install, but using: --unsafe-perm.

Summary:

1 - Download the ZooKeeper:
http://ftp.heanet.ie/mirrors/www.apache.org/dist/zookeeper/zookeeper-3.4.8/zookeeper-3.4.8.tar.gz

2 - Unzip it and add to the address:
(...)/node-zookeeper/deps/zookeeper

3 - Install the npm again:
sudo npm install --unsafe-perm

Thank you very much!
